velox/common/memory/MappedMemory.cpp

```diff
@@ -180,42 +218,43 @@ bool MappedMemoryImpl::allocate(
   throw std::runtime_error("Not implemented");
 }
 
-MachinePageCount MappedMemoryImpl::allocationSize(
+bool MappedMemoryImpl::allocateContiguous(
     MachinePageCount numPages,
-    MachinePageCount minSizeClass,
-    std::array<int32_t, kMaxSizeClasses>* sizeIndices,
-    std::array<int32_t, kMaxSizeClasses>* sizeCounts,
-    int32_t* numSizes) const {
-  int32_t needed = numPages;
-  int32_t pagesToAlloc = 0;
-  *numSizes = 0;
-  VELOX_CHECK_LE(
-      minSizeClass,
-      sizes_.back(),
-      "Requesting minimum size larger than largest size class");
-  for (int32_t sizeIndex = sizes_.size() - 1; sizeIndex >= 0; sizeIndex--) {
-    int32_t size = sizes_[sizeIndex];
-    bool isSmallest = sizeIndex == 0 || sizes_[sizeIndex - 1] < minSizeClass;
-    // If the size is less than 1/8 of the size from the next larger,
-    // use the next larger size.
-    if (size > (needed + (needed / 8)) && !isSmallest) {
-      continue;
-    }
-    int32_t numUnits = std::max(1, needed / size);
-    needed -= numUnits * size;
-    if (isSmallest && needed > 0) {
-      // If needed / size had a remainder, add one unit of smallest class.
-      numUnits++;
-      needed -= size;
+    Allocation* FOLLY_NULLABLE collateral,
+    ContiguousAllocation& allocation,
+    std::function<void(int64_t)> beforeAllocCB) {
+  MachinePageCount collateralSize = 0;
+  if (collateral) {
+    collateralSize = free(*collateral) / kPageSize;
+  }
+  auto largeCollateralSize = allocation.numPages();
```

Comment:
Variable names are confusing to me. collateral and allocation both represent already allocated memory that can be freed to make space for new allocation. Why is this memory in two "chunks"? Allocation* and ContiguousAllocation& ? It would be clearer to use 'collateral' in the name of both pieces.

Then, why allocation.numPages() is called largeCollateralSize and size of collateral is called collateralSize ? What makes contiguous allocation be "large"? Can we rename largeCollateralSize to numContinuousCollateralPages?